src/placeholders/placeholder.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+var util = require('./../util');
+
+/**
+ * Placeholder
+ *
+ * Representation of the placeholder to be placed over blank block elems
+ *
+ */
```

Comment:
document arguments

document life cycle

feel free to add "see <some other file> for details" to avoid writing the same things twice